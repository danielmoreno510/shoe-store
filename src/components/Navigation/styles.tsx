import {StyleSheet} from 'react-native';

import {primaryColor} from '../../constants';

export default StyleSheet.create({
  container: {
    flex: 1,
  },
  headerLeftContainer: {
    marginLeft: 12,
  },
  iconColor: {
    color: primaryColor,
  },
  headerRightContainer: {
    marginRight: 12,
  },
  badge: {
    marginBottom: -12,
    marginLeft: 14,
    height: 22,
    width: 22,
    zIndex: 1,
  },
  badgeText: {
    color: 'white',
  },
});
