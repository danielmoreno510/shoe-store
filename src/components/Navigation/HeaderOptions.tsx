import React from 'react';
import {TouchableOpacity, Text} from 'react-native';
import {Icon, Badge} from 'native-base';

import styles from './styles';

const HeaderOptions: React.FC<HeaderOptionsProps> = ({badge = 0, changeScreen, position}) => (
  <TouchableOpacity
    onPress={changeScreen}
    style={[position === 'left' ? styles.headerLeftContainer : styles.headerRightContainer]}>
    {!!badge && (
      <Badge warning style={styles.badge}>
        <Text style={styles.badgeText}>{badge}</Text>
      </Badge>
    )}
    <Icon
      type="MaterialIcons"
      name={position === 'left' ? 'menu' : 'shopping-cart'}
      style={styles.iconColor}
    />
  </TouchableOpacity>
);

export default HeaderOptions;
