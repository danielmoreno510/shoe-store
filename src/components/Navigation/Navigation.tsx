import React from 'react';
import {SafeAreaView} from 'react-native';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';

import HeaderOptions from './HeaderOptions';
import Home from '../../screens/Home';
import Menu from '../../screens/Menu';
import ShoesList from '../../screens/ShoesList';
import ShoeDetails from '../../screens/ShoeDetails';
import Cart from '../../screens/Cart';
import Pay from '../../screens/Pay';
import About from '../../screens/About';
import styles from './styles';
import {primaryColor} from '../../constants';

const Stack = createStackNavigator();
const screenOptions = {headerStyle: {shadowColor: primaryColor}, headerTintColor: primaryColor};

const Navigation: React.FC<NavigationProps> = ({cart}) => {
  const headerRight = ({navigation}: Navigation) => ({
    headerRight: () => (
      <HeaderOptions badge={cart.length} changeScreen={() => navigation.navigate('Cart')} position="right" />
    ),
  });

  const headerLeft = ({navigation}) => ({
    headerLeft: () => <HeaderOptions changeScreen={() => navigation.navigate('Menu')} position="left" />,
  });

  return (
    <NavigationContainer>
      <SafeAreaView style={styles.container}>
        <Stack.Navigator headerMode="screen" screenOptions={screenOptions}>
          <Stack.Screen
            name="Home"
            component={Home}
            options={({navigation}) => ({...headerLeft({navigation}), ...headerRight({navigation})})}
          />
          <Stack.Screen name="Shoes List" component={ShoesList} options={headerRight} />
          <Stack.Screen name="Shoe Detail" component={ShoeDetails} options={headerRight} />
          <Stack.Screen name="Cart" component={Cart} />
          <Stack.Screen name="Pay" component={Pay} />
          <Stack.Screen
            name="Menu"
            component={Menu}
            options={({navigation}) => ({
              ...headerLeft({navigation}),
              gestureDirection: 'horizontal-inverted',
            })}
          />
          <Stack.Screen name="About" component={About} />
        </Stack.Navigator>
      </SafeAreaView>
    </NavigationContainer>
  );
};

export default Navigation;
