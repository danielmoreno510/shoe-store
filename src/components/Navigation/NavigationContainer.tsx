import Navigation from './Navigation';
import {StoreService} from '../../redux/StoreService';

export const mapStateToProps = state => ({
  cart: state.shoes.cart,
});

const NavigationContainer = StoreService.connect(mapStateToProps)(Navigation);

export default NavigationContainer;
