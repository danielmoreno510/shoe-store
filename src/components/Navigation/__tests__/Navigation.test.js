import React from 'react';
import {shallow} from 'enzyme';

import Navigation from '../Navigation';

describe('Navigation component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      cart: [],
    };

    describe('WHEN the component is rendered', () => {
      const shallowWrapper = shallow(<Navigation {...testProps} />);
      shallowWrapper.find('Screen').at(0).props().options({navigation: jest.fn()});

      it('THEN should display a regular Navigation', () => {
        expect(shallowWrapper).toMatchSnapshot();
      });
    });
  });
});
