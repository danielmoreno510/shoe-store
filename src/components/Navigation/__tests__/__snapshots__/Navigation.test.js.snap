// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular Navigation 1`] = `
<ForwardRef(NavigationContainer)>
  <ForwardRef(SafeAreaView)
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <StackNavigator
      headerMode="screen"
      screenOptions={
        Object {
          "headerStyle": Object {
            "shadowColor": "#f1b30a",
          },
          "headerTintColor": "#f1b30a",
        }
      }
    >
      <Screen
        component={[Function]}
        name="Home"
        options={[Function]}
      />
      <Screen
        component={[Function]}
        name="Shoes List"
        options={[Function]}
      />
      <Screen
        component={[Function]}
        name="Shoe Detail"
        options={[Function]}
      />
      <Screen
        component={[Function]}
        name="Cart"
      />
      <Screen
        component={[Function]}
        name="Pay"
      />
      <Screen
        component={[Function]}
        name="Menu"
        options={[Function]}
      />
      <Screen
        component={[Function]}
        name="About"
      />
    </StackNavigator>
  </ForwardRef(SafeAreaView)>
</ForwardRef(NavigationContainer)>
`;
